docs/ensuring-repro/workflows/build-docker-gha.md

```diff
@@ -0,0 +1,31 @@
+# Automated Dockerfile building
+
+We use [GitHub Actions](https://docs.github.com/en/actions) (GHAs) to build Dockerfiles and push images to [Amazon ECR](https://aws.amazon.com/ecr/), a registry of containers such as Docker images.
+
+Docker building GHAs are automatically run in two circumstances:
+
+- When a pull request is filed with changes to any module's environment files, such as `renv.lock`, `conda.lock`, or the `Dockerfile` itself
+- When new `OpenScPCA-analysis` releases are made
+
+
+These pushed images are used in a variety of ways:
+
+- The `OpenScPCA-nf` workflow pulls module-specific Docker images to reproducibly run modules and generate results <!-- STUB_LINK openscpca-nf -->
+- [Module testing GHAs](./run-module-gha.md) will, as the given analysis module matures, pull the Docker-specific Docker image to create the environment used in the workflow
+- OpenScPCA contributors, as well as the wider research community, can freely pull module-specific images to reproducibly run OpenScPCA analysis modules, for example to locally run analysis modules or to develop within a Docker container
```

Comment:
I might remove the last clause here: we don't really talk at all about developing within containers, and I am not sure we want to raise it here. Seems distracting and potentially adds a bunch of questions we might rather not get into.